### DISCUSSION TITLE
nest and route behaviour

### DISCUSSION THREAD

**Discussion by ghost** [tokio-rs/axum#568] [Q&A] [answered]
### Version

v0.3.4

### Platform

Windows 11 64 bit

### Description

I have a GraphQL endpoint which I need to access from `/graphql` and `/graphql/` with below code:

```
let app = Router::new().nest("/graphql", post(graphql_handler));
```

 in previous version, both `/graphql` and `/graphql/`  work just fine pointing to the same graphql handler. after updating to 0.3 using `route` makes `/graphql/` unreachable while `nest` makes `/graphql` unreachable. is this expected? I need `/graphql/` because trunk proxy  somehow put the last slash to the request.

**Comment by jplatte** [comment, marked as answer]
If you don't need any sub-routes of the form `/graphql/foo` and want both `/graphql` and `/graphql/` to work *without redirects*, I think that would just be

```rust
Router::new()
    .route("/graphql", post(graphql_handler))
    .route("/graphql/", post(graphql_handler))
```

**Reply by davidpdrsn** [reply]
Yep thats correct.

By default axum will redirect to the route with/without trailing slash depending on which you've defined. So if you define `/foo` and request `/foo/` axum will redirect to `/foo`, or vice versa. If you define both routes explicitly then no redirects will happen.

**Reply by ghost** [reply]
thank you for the quick answer
